src/vs/platform/webContentExtractor/node/sharedWebContentExtractorService.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ISharedWebContentExtractorService } from '../common/webContentExtractor.js';
+import { URI } from '../../../base/common/uri.js';
+import { ResourceMap } from '../../../base/common/map.js';
+import { VSBuffer } from '../../../base/common/buffer.js';
+
+interface CacheEntry {
+	content: Uint8Array;
+	timestamp: number;
+}
+
+export class SharedWebContentExtractorService implements ISharedWebContentExtractorService {
+	_serviceBrand: undefined;
+	private _webContentsCache = new ResourceMap<CacheEntry>();
+	private readonly _cacheDuration = 24 * 60 * 60 * 1000; // 1 day in milliseconds
+
+	private isExpired(entry: CacheEntry): boolean {
+		return Date.now() - entry.timestamp > this._cacheDuration;
+	}
+
+	extractUrls(uris: URI): Promise<Uint8Array> {
```

Comment:
This should likely take a cancellation token in case the caller isn't interested in the result